### DISCUSSION TITLE
Repo-specific keybindings?

### DISCUSSION THREAD

**Discussion by macintacos** [jesseduffield/lazygit#2121] [Q&A]
I'm looking for a way to have a keybinding defined within a specific repository, but elsewhere fallback to a different shortcut. Is there a way to achieve this with lazygit? I was reading the documentation/wiki and couldn't find anything concrete.

**Comment by mark2185** [comment]
There is no such thing, currently. `lazygit` makes no distinction between different repositories.

Could you give us a little description of your use case, which things would you remap and why, just so we can see if others would want that.

**Reply by macintacos** [reply]
Sure! Specifically, I'm really just looking to be able to run a series of shell commands before attempting to commit with something like `commitizen`.

For example, some repositories I contribute to use `pre-commit`, while others just require ensuring that tests pass, so I would want to have the same key (in this case, for me I use `C`) run a command just before committing (which would presumably bail if it fails) instead of afterwards. Something like `command1 && git cz` in one repo and `command2 && git cz` in another.

Hopefully that makes sense. If there's a different approach I could use, I'm all ears!

**Reply by mark2185** [reply]
I haven't heard of `commitizen` before, so based on my extensive 5min googling adventure - wouldn't it make more sense to have a repo specific override for `git commit`, in the sense that it calls `commitizen` if necessary, otherwise continue as usual?

Found something like that [here](https://commitizen.github.io/cz-cli/) under `Optional: Running Commitizen on git commit`.

Then you would run `git commit` in every repo, and `git` would redirect you as needed.
